packages/commons/src/repositories/ReadModelRepository.ts

```diff
@@ -0,0 +1,32 @@
+import { EService, PersistentAgreement } from "pagopa-interop-models";
+import { Collection, MongoClient } from "mongodb";
+import { ReadModelDbConfig } from "../index.js";
+
+export type ReadModelRepository = {
+  eservices: Collection<{
+    data: EService | undefined;
+    metadata: { version: number };
+  }>;
+  agreements: Collection<{
+    data: PersistentAgreement;
+    metadata: { version: number };
+  }>;
+};
+
+export function readModelRepository({
+  readModelDbHost: host,
+  readModelDbPort: port,
+  readModelDbUsername: username,
+  readModelDbPassword: password,
+  readModelDbName: database,
+}: ReadModelDbConfig): ReadModelRepository {
+  const mongoDBConnectionURI = `mongodb://${username}:${password}@${host}:${port}`;
+  const client = new MongoClient(mongoDBConnectionURI, {
```

Comment:
does the client handle internally any singleton for the connection pool? Or it creates a new connection for each client?
My concern is to avoid that multiple invocation of this function (eg if we have several services files) will open more than one connection